source/agora/consensus/data/Block.d

```diff
@@ -434,11 +434,12 @@ unittest
     Params:
         prev_block = the previous block
         txs = the transactions that will be contained in the new block
+        enrollments = the enrollments that will be contained in the new block
 
 *******************************************************************************/
 
 public Block makeNewBlock (Transactions)(const ref Block prev_block,
-    Transactions txs) @safe nothrow
+    Transactions txs, Enrollment[] enrollments = null) @safe nothrow
```

Comment:
Did you made it a default parameter so you don't have to change a lot of unittest code?

In general, we should not design our functions so they are specialized / convenient for existing unittest code. So I would not make this a default parameter. The only exception to that is for `version (unittest)` functions, which are only used as helper functions for unittest code.

Reply:
The intention of this is to assume that there may be no enrollments.
This is because most blocks don't have many cases.

Reply:
This function is only called from a single place in Agora (I'm not talking about unittest code), and that's in the `Ledger`.  And you directly pass `his.enroll_man.getUnregisteredEnrollments(enrollments)`. So the default parameter is not necessary.

Reply:
Yes, the default parameter is not necessary.
I'll have to go back to my first commit.
I had modified many unit test codes. :)